### PR TITLE
Support special x-req mime type in script elements.

### PR DESCRIPTION
**Pull request by jjrv**

### CODE REVIEW

File: src/Loader.ts [hunk 1]
```diff
@@ -292,19 +298,16 @@ export class Loader implements LoaderPlugin {
 					err.message += '\n    translating ' + record.resolvedKey;
 				}
 				throw err;
-			});
+			}).then(() => Promise.all(record.depList.map(
```
Comment (ISNIT0) [inline src/Loader.ts:301]: I generally stay away from multi-line arrow functions with no `{}`s
Not a big deal, but I think this makes things a little more complex to read

File: src/plugin/Document.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	discover(record: Record) {
+		let num = 1;
+
+		for(let element of [].slice.call(getTags && getTags('script')) as HTMLScriptElement[]) {
+			const type = element.type;
+			if(type && type.substr(0, 5) == 'x-req') {
+				if(element.src) {
+					record.addDep(URL.resolve(origin, element.src));
+				} else {
+					record.addDep(URL.resolve(origin, '#' + num++ + '.js'), {
```
Comment (ISNIT0) [inline src/plugin/Document.ts]: Yuuuuck, there are a lot of plusses here, and a non-obvious mutation of `num`

File: src/plugin/NodeResolve.ts [hunk 3]
```diff
@@ -441,7 +464,7 @@ export class NodeResolve implements LoaderPlugin {
 				if(!pkg) return Promise.reject(new Error('Error fetching ' + resolvedKey));
 
 				resolvedKey = loader.resolveSync(key, baseKey, ref);
-				return checkFile(loader, resolvedKey, key, baseKey, ref);
+				return checkFile(loader, resolvedKey, key, baseKey, ref!);
```
Comment (ISNIT0) [inline src/plugin/NodeResolve.ts]: There seem to be a lot of `var!`s, this is a bit of a code smell, perhaps we should update they types to actually represent the state?

**Review by ISNIT0 — CHANGES_REQUESTED** [review]

File: src/plugin/NodeResolve.ts (hunk 3 above)
Comment (jjrv) [reply src/plugin/NodeResolve.ts]: The compiler annoyingly fails to apply some type inference info from variables in outer functions to their references in inner functions. Fixed by adding default values to parameters at the cost of a slightly smaller minified result.

File: src/Loader.ts (hunk 1 above)
Comment (jjrv) [reply src/Loader.ts:301]: Here the block would still only contain a return statement...

File: src/plugin/Document.ts (hunk 2 above)
Comment (jjrv) [reply src/plugin/Document.ts]: Good point, better make the mutation more explicit.

**Pull request merged**